libs/community/langchain_community/document_loaders/onedrive.py

```diff
@@ -1,94 +1,16 @@
-"""Loads data from OneDrive"""
+from typing import Any
 
-from __future__ import annotations
-
-import logging
-from typing import TYPE_CHECKING, Iterator, List, Optional, Sequence, Union
-
-from langchain_core.documents import Document
 from pydantic import Field
 
-from langchain_community.document_loaders.base_o365 import (
-    O365BaseLoader,
-    _FileType,
-)
-from langchain_community.document_loaders.parsers.registry import get_parser
-
-if TYPE_CHECKING:
-    from O365.drive import Drive, Folder
-
-logger = logging.getLogger(__name__)
+from langchain_community.document_loaders import SharePointLoader
 
 
-class OneDriveLoader(O365BaseLoader):
-    """Load from `Microsoft OneDrive`."""
+class OneDriveLoader(SharePointLoader):
+    """Load documents from OneDrive. Uses `SharePointLoader` under the hood."""
```

Comment:
why remove `Microsoft`?

Reply:
I rewrote the whole comment and ended up leaving out "Microsoft". I can put that back.

Reply:
Done